src/test/unit/tuvx/tuvx_c_api.cpp

```diff
@@ -385,3 +385,340 @@ TEST_F(TuvxCApiTest, CanCreateProfileMap)
   ASSERT_TRUE(IsSuccess(error));
   DeleteError(&error);
 }
+
+TEST_F(TuvxCApiTest, CannotGetConfiguredRadiator)
+{
+  const char* yaml_config_path = "examples/ts1_tsmlt.yml";
+  SetUp(yaml_config_path);
+  Error error;
+  RadiatorMap* radiator_map = GetRadiatorMap(tuvx, &error);
+  ASSERT_TRUE(IsSuccess(error));
+  ASSERT_NE(radiator_map, nullptr);
+  Radiator* radiator = GetRadiator(radiator_map, "foo", &error);
+  ASSERT_FALSE(IsSuccess(error));  // non-host grid
+  ASSERT_EQ(radiator, nullptr);
+  DeleteRadiatorMap(radiator_map, &error);
+  ASSERT_TRUE(IsSuccess(error));
+  DeleteError(&error);
+}
+
+TEST_F(TuvxCApiTest, CanCreateRadiator)
+{
+  Error error;
+  Grid* height = CreateGrid("height", "km", 3, &error);
+  Grid* wavelength = CreateGrid("wavelength", "nm", 2, &error);
+  Radiator* radiator = CreateRadiator("foo", height, wavelength, &error);
+  ASSERT_TRUE(IsSuccess(error));
+  ASSERT_NE(radiator, nullptr);
+
+  // Test for optical depths
+  std::size_t num_vertical_layers = 3;
+  std::size_t num_wavelength_bins = 2;
+  // Allocate array as 1D
+  double* optical_depths_1D = new double[num_wavelength_bins * num_vertical_layers];
+  // Allocate an array of pointers to each row
+  double** optical_depths = new double*[num_vertical_layers];
+  // Fill in the pointers to the rows
+  for (int row = 0; row < num_vertical_layers; row++)
+  {
+    optical_depths[row] = &optical_depths_1D[row * num_wavelength_bins];
+  }
+  int i = 1;
+  for (int row = 0; row < num_vertical_layers; row++)
+  {
+    for (int col = 0; col < num_wavelength_bins; col++)
+    {
+      optical_depths[row][col] = 10 * i;
+      i++;
+    }
+  }
```

Comment:
Do we like this? I initially wanted to use `vector` but couldn't find a way to make it work for the fortran APIs to be able to index the array components. (e,g., `optical_depths[row][col]`) @mattldawson, @K20shores, any thoughts on this implementation?

Reply:
I like this solution! This seems like it will work well with the fortran wrapper, and it keeps the data contiguous in memory